agglayer/grpc/agglayer_grpc_client.go

```diff
@@ -3,24 +3,36 @@ package grpc
 import (
 	"context"
 	"errors"
-	"strings"
+	"fmt"
+	"time"
 
 	node "buf.build/gen/go/agglayer/agglayer/grpc/go/agglayer/node/v1/nodev1grpc"
 	v1nodetypes "buf.build/gen/go/agglayer/agglayer/protocolbuffers/go/agglayer/node/types/v1"
 	v1 "buf.build/gen/go/agglayer/agglayer/protocolbuffers/go/agglayer/node/v1"
 	v1types "buf.build/gen/go/agglayer/interop/protocolbuffers/go/agglayer/interop/types/v1"
 	"github.com/agglayer/aggkit/agglayer/types"
 	"github.com/agglayer/aggkit/bridgesync"
-	aggkitCommon "github.com/agglayer/aggkit/common"
+	aggkitcommon "github.com/agglayer/aggkit/common"
 	treetypes "github.com/agglayer/aggkit/tree/types"
 	"github.com/ethereum/go-ethereum/common"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 var (
 	errUndefinedAggchainData = errors.New("undefined aggchain data parameter")
 	errUnknownAggchainData   = errors.New("unknown aggchain data type")
 )
 
+const (
+	// maxRequestRetries is the maximum number of retries for gRPC requests
+	// before giving up and returning an error
+	maxRequestRetries = 8
```

Comment:
Maybe it makes sense, like @joanestebanr mentioned to move this to the config.

I would probably have some common `GRPC` config part in the `aggkit.toml`:

```
[GRPC]
MaxRequestRetries = 8
RetryInitialDelay = "1s"
MinConnectTimeout = "5min"
```

Reply:
Fixed https://github.com/agglayer/aggkit/pull/524/commits/f32ebab25e24f4e3bbd0d9825c63dd99182dc85d